src/utils/stat.js

```diff
@@ -214,13 +214,9 @@ export const getStatData = (data, type, opts = {}) => {
           input: {
             id: 'weight',
             label: t('Weight'),
-            suffix: {
-              id: 'units',
-              options: dailyDoseUnitOptions,
-              value: opts.suffixValue || dailyDoseUnitOptions[0],
-            },
+            suffix: data.weight ? (data.weight.units ? (data.weight.units) : 'kg' ) : 'kg',
```

Comment:
Simplify by:
```suggestion
            suffix: _.get(data, 'weight.unit', 'kg'),
```
Note: Typo in portal-api for the units: it's `unit`.